src/parquet/util/memory.h

```diff
@@ -340,6 +340,12 @@ class PARQUET_EXPORT InMemoryOutputStream : public OutputStream {
 
   virtual void Write(const uint8_t* data, int64_t length);
 
+  // Clears the stream
+  void Clear() { size_ = 0; }
+
+  // Get pointer to the underlying buffer
+  std::shared_ptr<Buffer> GetBufferRef() const { return buffer_; }
```

Comment:
I don't require the caller to share ownership in my current patch. Will change it.